marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/default/lib.sjs

```diff
@@ -15,12 +15,33 @@ function getMappingWithVersion(mappingName, version) {
   return fn.head(cts.search(cts.andQuery([cts.collectionQuery('http://marklogic.com/data-hub/mappings'), cts.jsonPropertyValueQuery('name', mappingName, ['case-insensitive']), cts.jsonPropertyValueQuery('version', version)])));
 }
 
+function getSourceContext(sourceContext) {
+  let connector = "/*:";
+  let srcCtxArr;
+
+  sourceContext = sourceContext.startsWith("/") ? sourceContext.substring(1,sourceContext.length) : sourceContext;
+  srcCtxArr = sourceContext.split("/");
+  sourceContext = "";
+
+  srcCtxArr.forEach(function(element) {
+    if(element.indexOf(':') === -1) {
+      sourceContext += connector + element;
+    } else {
+      sourceContext += "/" + element;
+    }
+  });
+  return sourceContext;
+}
+
 function processInstance(model, mapping, content) {
- return  extractInstanceFromModel(model, model.info.title, mapping, content);
+ return extractInstanceFromModel(model, model.info.title, mapping, content);
```

Comment:
what if there's no info or info.title present? Probably want to do an existence check before we throw an exception here.

Reply:
We can't deploy an entity without the info object in {entity-name}.entity.json file. It will throw the following exception.
Local message: failed to apply resource at documents: Internal Server Error. Server Message: TDE-INDEX: Error applying template entity-types-info-json.tdex to document /entities/Order2.entity.json: TDE-EVALFAILED: Eval for Object='$title' returns an empty sequence . See the MarkLogic server error log for further detail.

Should we still have this check.

Reply:
@aebadirad Thanks for letting know that TDE's can be turned off. I will add the check